api/build.gradle

```
@@ -10,12 +13,63 @@ repositories {
 }
 
 dependencies {
+    implementation project(path: ':app')
     testImplementation 'org.jetbrains.kotlin:kotlin-test'
+    implementation 'org.springdoc:springdoc-openapi-starter-webmvc-ui:2.5.0'	//Swagger
+    implementation platform("com.fasterxml.jackson:jackson-bom:2.15.2") // 원하는 Jackson 버전으로 변경
+    implementation "com.fasterxml.jackson.core:jackson-core"
+    implementation "com.fasterxml.jackson.core:jackson-databind"
+    implementation "com.fasterxml.jackson.core:jackson-annotations"
 }
 
 test {
     useJUnitPlatform()
 }
+
 kotlin {
     jvmToolchain(17)
-}
\ No newline at end of file
+}
+
+// OpenAPI Generator 설정
+def openApiGeneratedDir = "$projectDir" // api 모듈 내부 docs 패키지에 생성된 코드 저장
+def contractDir = "$projectDir/docs/contract" // OpenAPI 사양 파일 디렉터리
+
+// 모든 OpenAPI Generator 태스크를 구성 단계에서 등록
+def openApiTasks = fileTree(dir: contractDir, include: ['*.yaml', '*.yml']).collect { yamlFile ->
+    def fileNameWithoutExtension = yamlFile.name.replaceFirst(/\.[^\.]+$/, '') // 확장자 제거
+
+    tasks.register("openApiGenerate_${fileNameWithoutExtension}", GenerateTask) {
+        generatorName.set("kotlin-spring")
+        inputSpec.set("$contractDir/${yamlFile.name}")
```

```suggestion
sout('hello world')
```

suggestion 안한지 오래된 거 같아서 해봤습니다!!